lib/gui/etcher.js

```diff
@@ -18,14 +18,39 @@
 
 const electron = require('electron')
 const path = require('path')
+const _ = require('lodash')
+const { autoUpdater } = require('electron-updater')
+const Bluebird = require('bluebird')
 const EXIT_CODES = require('../shared/exit-codes')
 const buildWindowMenu = require('./menu')
 const settings = require('./app/models/settings')
-
+const analytics = require('./app/modules/analytics')
+const { getConfig } = require('../shared/utils')
 /* eslint-disable lodash/prefer-lodash-method */
 
 const config = settings.getDefaults()
 
+const configUrl = settings.get('configUrl') || 'http://balena.io/etcher/static/config.json'
+
+/**
+ *
+ * @param {Number} timer - timer to wait to check for updates
+ * @example checkForUpdates()
+ */
+const checkForUpdates = async (timer) => {
+  while (true) {
+    try {
+      const release = await autoUpdater.checkForUpdates()
+      if (release.updateInfo.stagingPercentage) {
+        await autoUpdater.downloadUpdate(release.cancellationToken)
```

Comment:
Why is `release.cancellationToken` needed?

Reply:
It's used to cancel the download promise by `electron-updater`